responses/tests/test_recorder.py

```diff
@@ -21,42 +21,141 @@ def get_data(host, port):
         "responses": [
             {
                 "response": {
-                    "method": "GET",
-                    "url": f"http://{host}:{port}/404",
+                    "auto_calculate_content_length": False,
                     "body": "404 Not Found",
-                    "status": 404,
                     "content_type": "text/plain",
-                    "auto_calculate_content_length": False,
+                    "matchers": [
+                        {
+                            "query_param_matcher": {
+                                "args": {"params": {}, "strict_match": True},
+                                "matcher_import_path": "responses.matchers",
+                            }
+                        },
+                        {
+                            "header_matcher": {
+                                "args": {
+                                    "headers": {
+                                        "Accept": "*/*",
+                                        "Accept-Encoding": "gzip, deflate",
+                                        "Connection": "keep-alive",
+                                        "User-Agent": "python-requests/2.30.0",
+                                    },
+                                    "strict_match": False,
+                                },
+                                "matcher_import_path": "responses.matchers",
```

Comment:
hi @markstory 

pls have a look on the draft of the recorder.
I have a question/concern how to proceed with replay

considering that we need to support custom matchers, we probably want to import matchers as well.
I found a way how to extract the location of the mather. however, then we need to do import during runtime and local to each function, which does not sound really smart

what could be the options?

I think asking users to import required matchers defeats the purpose of record/replay

Reply:
Hrm, importing userland code is possible but it could get messy. I think we're also taking a risk that userland matcher parameters will be 'serializable' into JSON, which isn't guaranteed. We could require that in order to use recording :shrug: